openapipulschema/provider.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openapipulschema
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	pulschema "github.com/cloudy-sky-software/pulschema/pkg"
+	"github.com/getkin/kin-openapi/openapi3"
+	p "github.com/pulumi/pulumi-go-provider"
+	"github.com/pulumi/pulumi-go-provider/middleware/schema"
+	pschema "github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+)
+
+const packageName = "openapi"
+
+func Provider(oaURL url.URL, metadata schema.Metadata) (*p.Provider, error) {
+	openAPILoader := openapi3.NewLoader()
+	openAPILoader.IsExternalRefsAllowed = true
+
+	openAPIDoc, err := openAPILoader.LoadFromURI(&oaURL)
+	if err != nil {
+		return nil, fmt.Errorf("loading OpenAPI spec at %v: %w", oaURL, err)
+	}
+	openAPIDoc.InternalizeRefs(context.Background(), nil)
+
+	pkg := pschema.PackageSpec{
+		Name: packageName,
```

Comment:
Providers are not allowed to know their name until they are run, so this *must* be a temporary value. Can we just leave it blank?